### PR TITLE
Fix: PHP errors caused by deprecation module [ED-8650]

### LINKED ISSUE #20062: 🐞 Bug Report: Fatal errors when saving "Site Settings". PHP 8.0 (🔃 Has PR)
### Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest stable version of Elementor.

### Description

Hi! I faced some issues with Elementor + Elementor PRO + PHP 8.0. Saving "Site Setting" causes PHP Fatal errors.

```
Uncaught TypeError: Cannot access offset of type string on string in /wp-content/plugins/elementor-pro/plugin.php:350
Stack trace:
/wp-includes/class-wp-hook.php(309): ElementorPro\Plugin->get_responsive_stylesheet_templates('...')
/wp-includes/plugin.php(235): WP_Hook->apply_filters(NULL, Array)
/wp-content/plugins/elementor/modules/dev-tools/deprecation.php(357): apply_filters_ref_array('elementor/core/...', Array)
/wp-content/plugins/elementor/core/breakpoints/manager.php(533): Elementor\Modules\DevTools\Deprecation->apply_deprecated_filter('elementor/core/...', Array, '3.2.0', 'elementor/core/...')
/wp-content/plugins/elementor/core/breakpoints/manager.php(420): Elementor\Core\Breakpoints\Manager::get_stylesheet_templates()
/wp-content/plugins/elementor/core/kits/documents/tabs/settings-layout.php(292): Elementor\Core\Breakpoints\Manager::compile_stylesheet_templates()
/wp-content/plugins/elementor/core/kits/documents/kit.php(98): Elementor\Core\Kits\Documents\Tabs\Settings_Layout->on_save(Array)
[internal function]: Elementor\Core\Documents_Manager->ajax_save(Array, Object(Elementor\Core\Common\Modules\Ajax\Module))
/wp-content/plugins/elementor/core/common/modules/ajax/module.php(171): call_user_func(Array, Array, Object(Elementor\Core\Common\Modules\Ajax\Module))
/wp-includes/class-wp-hook.php(307): Elementor\Core\Common\Modules\Ajax\Module->handle_ajax_request('')
/wp-includes/class-wp-hook.php(331): WP_Hook->apply_filters('', Array)
/wp-includes/plugin.php(476): WP_Hook->do_action(Array)
/wp-admin/admin-ajax.php(187): do_action('wp_ajax_element...')
{main} thrown in /wp-content/plugins/elementor-pro/plugin.php on line 350
```

### Steps to reproduce

1) Open Elementor "Site Settings"
2) Add global color
3) Save

### Isolating the problem

- [X] This bug happens with only Elementor plugin active (and Elementor Pro).
- [X] This bug happens with a Blank WordPress theme active ([Hello theme](https://wordpress.org/themes/hello-elementor/)).
- [X] I can reproduce this bug consistently using the steps above.

### System Info

WordPress - 6.0.2
Elementor - 3.7.8
Elementor Pro - 3.7.3
PHP - 8.0.24

### PR DESCRIPTION
**Pull request by gerasimovdaniel**
## PR Checklist

- [x] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

* Fix. PHP errors caused by deprecation module

## Description
An explanation of what is done in this PR

* Deprecation module method [apply_deprecated_filter](https://github.com/elementor/elementor/blob/97f2db9d04a34dd3b1fc9fc101ab2536d6834528/modules/dev-tools/deprecation.php#L350) uses [apply_filters_ref_array](https://developer.wordpress.org/reference/functions/apply_filters_ref_array/) which expect the second argument to be an array of filter arguments, but in [/core/breakpoints/manager.php](https://github.com/elementor/elementor/blob/master/core/breakpoints/manager.php#L533)  actual value is given. It causes PHP errors when "Site Settings" is saved. In PHP 8.0+ it causes fatal errors.

## Test instructions
This PR can be tested by following these steps:

* Activate Elementor Pro
* Add global color in "Site Settings"
* Save

## Quality assurance

- [x] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #20062

### CODE REVIEW

**Comment by gerasimovdaniel** [comment]
@matipojo @KingYes is there anything I can do to get a PR review?

File: modules/dev-tools/deprecation.php [hunk 1]
```diff
@@ -349,7 +349,7 @@ public function do_deprecated_action( $hook, $args, $version, $replacement = '',
 	 */
 	public function apply_deprecated_filter( $hook, $args, $version, $replacement = '', $base_version = null ) {
```
Comment (StyleShit) [inline modules/dev-tools/deprecation.php:350]:
```suggestion
	public function apply_deprecated_filter( $hook, array $args, $version, $replacement = '', $base_version = null ) {
```

File: modules/dev-tools/deprecation.php [hunk 2]
```diff
@@ -349,7 +349,7 @@ public function do_deprecated_action( $hook, $args, $version, $replacement = '',
...
 		if ( ! has_action( $hook ) ) {
-			return $args;
+			return $args[0];
```
Comment (StyleShit) [inline modules/dev-tools/deprecation.php:352]:
```suggestion
			return $args[0] ?? null;
```

**Review by StyleShit — CHANGES_REQUESTED** [review]

File: modules/dev-tools/deprecation.php (hunk 1 above)
Comment (StyleShit) [inline modules/dev-tools/deprecation.php:350]:
```suggestion
	public function apply_deprecated_filter( $hook, array $args, $version, $replacement = '', $base_version = null ) {
```

File: modules/dev-tools/deprecation.php (hunk 2 above)
Comment (StyleShit) [inline modules/dev-tools/deprecation.php:352]:
```suggestion
			return $args[0] ?? null;
```

**Review by StyleShit — CHANGES_REQUESTED** [review]

File: modules/dev-tools/deprecation.php [hunk 3]
```diff
@@ -347,9 +347,9 @@ public function do_deprecated_action( $hook, $args, $version, $replacement = '',
 	 * @return mixed
 	 * @throws \Exception
 	 */
-	public function apply_deprecated_filter( $hook, $args, $version, $replacement = '', $base_version = null ) {
+	public function apply_deprecated_filter( $hook, array $args, $version, $replacement = '', $base_version = null ) {
```
Comment (StyleShit) [inline modules/dev-tools/deprecation.php:350]:
```suggestion
	public function apply_deprecated_filter( $hook, $args, $version, $replacement = '', $base_version = null ) {
```

**Review by StyleShit — CHANGES_REQUESTED** [review]

File: modules/dev-tools/deprecation.php [hunk 4]
```diff
@@ -349,7 +349,7 @@ public function do_deprecated_action( $hook, $args, $version, $replacement = '',
 	 */
 	public function apply_deprecated_filter( $hook, $args, $version, $replacement = '', $base_version = null ) {
 		if ( ! has_action( $hook ) ) {
-			return $args;
+			return $args[0] ?? null;
```
Comment (StyleShit) [inline modules/dev-tools/deprecation.php:352]:
```suggestion
			// `$args` should be an array, but in order to keep BC, we need to support non-array values.
			if ( is_array( $args ) ) {
				return $args[0] ?? null;
			}

			return $args;
```

**Review by StyleShit — CHANGES_REQUESTED** [review]

**Review by StyleShit — APPROVED** [review]

**Comment by matipojo** [comment]
Thanks @gerasimovdaniel !

**Pull request merged**